src/main/scala/backgammon/Situation.scala

```diff
@@ -46,10 +65,74 @@ case class Situation(board: Board, player: Player) {
 
   def canOnlyEndTurn: Boolean = canEndTurn && !canMove && !canDrop && !canLift
 
+  private def forcedEndTurn = canOnlyEndTurn && board.history.forcedTurn
+
   def canUseDice: Boolean = board.unusedDice.nonEmpty && (canMove || canDrop || canLift)
 
   def canUndo: Boolean = board.history.lastAction.map(_.undoable).getOrElse(false)
 
+  def canCapture: Boolean = actions
+    .map {
+      case m: Move => m.capture.nonEmpty
+      case d: Drop => d.capture.nonEmpty
+      case _       => false
+    }
+    .contains(true)
+
+  private def commonSetElements[A](sets: List[Set[A]]): Set[A] =
+    sets.fold(sets.headOption.getOrElse(Set())) { (a, b) => a intersect b }
+
+  // forcedPair should not be true if the order matters (i.e. 3/2 captures, but 2/3 doesn't)
+  lazy val forcedPair: Boolean =
+    if (actions.length == 2 && board.unusedDice.toSet.size == 2)
+      actions
+        .flatMap { a =>
+          a.lazySituationAfter.actions.map { a2 =>
+            (a2.lazySituationAfter.board.pieces, a2.lazySituationAfter.board.pocketData)
+          }
+        }
+        .toSet
+        .size == 1
+    else false
+
+  // If orig is in all paths of validTurns then we know that a piece on that pos has to
+  // play at some point during the turn and there will only be one forced action this turn.
+  // The forced action is the action which gives the most paths
+  // This assumes that some of the other forced action checks have been done first,
+  // and that there is not a capture possible in any of the current available actions
+  private lazy val forcedSingle: Option[Action] = // None
```

Comment:
what would happen here if you had 2 remaining pieces on the 3 and 2 spots and rolled a 2 and 3. Can you write a test for that easily (is from position available?). I think it might do something unexpected just reading through the code but a second pair of eyes would be good.